js/src/util/args.ts

```diff
@@ -21,11 +21,26 @@ import { Column } from '../column';
 import { Vector } from '../vector';
 import { DataType } from '../type';
 import { Chunked } from '../vector/chunked';
+import { BigIntArray, TypedArray } from '../interfaces';
+import { FloatVector, IntVector } from '../vector/index';
 
 type RecordBatchCtor = typeof import('../recordbatch').RecordBatch;
 
 const isArray = Array.isArray;
 
+/** @ignore */
+export function isTypedArray(arr: any): arr is TypedArray | BigIntArray {
+    return ArrayBuffer.isView(arr) && 'BYTES_PER_ELEMENT' in arr;
```

Comment:
I think this should be enough. Otherwise, we can use something like https://underscorejs.org/docs/modules/isTypedArray.html.

Reply:
Arquero just uses `instanceof TypedArray`. 

https://github.com/uwdata/arquero/blob/af843a4377254bab970d40559879f9191fe81079/src/util/is-typed-array.js#L4